src/ddscxx/include/dds/core/policy/CorePolicy.hpp

```diff
@@ -152,6 +152,9 @@ UserData;
 typedef dds::core::policy::detail::WriterDataLifecycle
 WriterDataLifecycle;
 
+typedef dds::core::policy::detail::WriterBatching
+WriterBatching;
```

Comment:
After all the effort spent by the OMG in defining the API to allow vendor-specific extensions accessible only via the `->` operator and presumably with the intent to avoid the need to modify the OMG-provided header files, I wonder if it is necessary (and if not, if it is best) to add the new QoS here rather than in an implementation-specific file.

It looks to me like OpenSplice managed to add a "ReaderLifespan" QoS setting [in the delegate](https://github.com/ADLINK-IST/opensplice/blob/a4818af5f5931834501055cd7ce2339de4f0af13/src/api/dcps/isocpp2/include/org/opensplice/core/policy/PolicyDelegate.hpp#L663) without modifying the OMG header files for it, other than a comment and (shockingly) a need to maintain a [copy](https://github.com/ADLINK-IST/opensplice/blob/a4818af5f5931834501055cd7ce2339de4f0af13/src/api/dcps/isocpp2/include/dds/core/detail/ProprietaryApi.hpp#L969) of proprietary stuff just for [generating documentation](https://github.com/ADLINK-IST/opensplice/blob/a4818af5f5931834501055cd7ce2339de4f0af13/src/api/dcps/isocpp2/include/dds/core/detail/ProprietaryApi.hpp#L53) ...

To me, that solution is worse than simply adding it to the OMG header files. But perhaps you could do me a favour and have a look whether my conclusion is correct and whether you agree with my sentiment that your choice is likely the lesser of the evils?

(I have to admit to being torn between trying to follow what the OMG did, and throwing out most of it and just having a simpler structure with fewer levels of indirection in the templates.)

Reply:
It appears that the two of us are in agreement. Surely that means we are right 🤣 Merging it before anyone complains!